### PR TITLE
Disable version patching in the CI run

### PR DESCRIPTION
**Pull request by NextTurn**
Work item #378 

Please add `patch: true` in release job configuration.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -20,7 +19,7 @@ before_build:
   - cmd: '"C:\Program Files (x86)\Microsoft SDKs\Windows\v10.0A\bin\NETFX 4.8 Tools\sn.exe" -k winsw_key.snk'
 
 dotnet_csproj:
-  patch: true
+  # patch: false
```
Comment (oleg-nenashev) [inline appveyor.yml]: It is not going to fix the issue, because we still need to patch the release version. I will try to find alternate way

Comment (oleg-nenashev) [reply appveyor.yml]: TL;DR: UI does not override this setting AFAICT

**Comment by oleg-nenashev** [comment]
As a quick fix, I will just duplicate the YAML config

File: appveyor.yml (hunk 1 above)
Comment (NextTurn) [reply appveyor.yml]: > UI does not override this setting AFAICT

Then move the entire `dotnet_csproj:` sections into release configuration?

**Comment by NextTurn** [comment]
PR builds don't need version patching. Is it possible to move the patching task to the release job?

**Comment by oleg-nenashev** [comment]
Yes, will do

**Comment by oleg-nenashev** [comment]
According to my experiments with #380 , copy-pasting files does not work when appveyor.yml is present. I will remove the version field and setup it's injection in the CI job, but I will have to keep the version patch for now

**Comment by NextTurn** [comment]
So you've set up different version patterns for release and non-release jobs and no action is required for the current *appveyor.yml* file?

**Comment by oleg-nenashev** [comment]
> So you've set up different version patterns for release and non-release jobs and no action is required for the current appveyor.yml file?

I have set up a temporary solution, only that. I am not happy about different versioning patterns which may confuse users at some point. I am also not happy about having to store configs in job settings instead of configuration as code. But at least it restores the status quo we had before 2.4.0

**Comment by casz** [comment]
I'd suggest looking into using GitVersion.
See https://github.com/nmklotas/GitLabApiClient the main csproj and the appveyor.yml is of interest.
I can link to other projects with GitVersion in an automated workflow.

GitLabApiClient is fully automated for releasing to nuget.org with versioning when a tag is added.

**Pull request merged**